sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClientBuilder.java

```diff
@@ -292,9 +293,17 @@ private HttpPipeline createHttpPipeline() {
         HttpPolicyProviders.addBeforeRetryPolicies(policies);
         policies.add(ClientBuilderUtil.validateAndGetRetryPolicy(retryPolicy, retryOptions, new RetryPolicy()));
         policies.add(new AddDatePolicy());
-        if (azureKeyCredential != null) {
-            policies.add(new AzureKeyCredentialPolicy("api-key", azureKeyCredential));
+        policies.add(new CookiePolicy());
+        if (keyCredential != null) {
+            KeyCredentialPolicy keyCredentialPolicy;
+            if (endpoint != null) {
```

Comment:
Public OpenAI service don't have an endpoint. 
If use provides `endpoint`, we assume they are targeting the Azure OpenAI service.
Else, targeting OpenAI service.